aws_lambda_powertools/event_handler/openapi/swagger_ui/oauth2.py

```diff
@@ -0,0 +1,144 @@
+# ruff: noqa: E501
+from typing import Dict, Optional, Sequence
+
+from pydantic import BaseModel, Field
+
+from aws_lambda_powertools.event_handler.openapi.pydantic_loader import PYDANTIC_V2
+
+
+# Based on https://swagger.io/docs/open-source-tools/swagger-ui/usage/oauth2/
+class OAuth2Config(BaseModel):
+    """
+    OAuth2 configuration for Swagger UI
+    """
+
+    # The client ID for the OAuth2 application
+    clientId: str = Field(alias="client_id")
+
+    # The realm in which the OAuth2 application is registered. Optional.
+    realm: Optional[str] = Field(default=None)
+
+    # The name of the OAuth2 application
+    appName: str = Field(alias="app_name")
+
+    # The scopes that the OAuth2 application requires. Defaults to an empty list.
+    scopes: Sequence[str] = Field(default=[])
+
+    # Additional query string parameters to be included in the OAuth2 request. Defaults to an empty dictionary.
+    additionalQueryStringParams: Dict[str, str] = Field(alias="additional_query_string_params", default={})
+
+    # Whether to use basic authentication with the access code grant type. Defaults to False.
+    useBasicAuthenticationWithAccessCodeGrant: bool = Field(
+        alias="use_basic_authentication_with_access_code_grant",
+        default=False,
+    )
+
+    # Whether to use PKCE with the authorization code grant type. Defaults to False.
+    usePkceWithAuthorizationCodeGrant: bool = Field(alias="use_pkce_with_authorization_code_grant", default=False)
+
+    if PYDANTIC_V2:
+        model_config = {"extra": "allow"}
+    else:
+
+        class Config:
+            extra = "allow"
+            allow_population_by_field_name = True
+
+
+class OAuth2UnsafeConfig(OAuth2Config):
+    """
+    This class extends the OAuth2Config class and includes the client secret.
+    This class NEVER BE USED IN PRODUCTION as it will expose sensitive information.
+    """
+
+    # The client secret for the OAuth2 application. This is sensitive information.
+    clientSecret: str = Field(alias="client_secret")
```

Comment:
Move this to the parent class and add a validator to check for the presence of `POWERTOOLS_DEV`.